CondCore/ESSources/plugins/CondDBESSource.cc

```diff
@@ -123,6 +125,17 @@ CondDBESSource::CondDBESSource(const edm::ParameterSet& iConfig)
     globaltag = iConfig.getParameter<std::string>("globaltag");
     // the global tag _requires_ a connection string
     m_connectionString = iConfig.getParameter<std::string>("connect");
+
+    if (!globaltag.empty()) {
+      edm::Service<edm::SiteLocalConfig> siteLocalConfig;
+      if (siteLocalConfig.isAvailable()) {
+        std::string const& localConnectPrefix = siteLocalConfig->localConnectPrefix();
+        std::string const& localConnectSuffix = siteLocalConfig->localConnectSuffix();
+        if (!localConnectPrefix.empty() && !localConnectSuffix.empty()) {
```

Comment:
Should the suffix be allowed to be empty?

How about `SiteLocalConfig` would have a function telling if the `local-config` was specified or not, and then using the prefix and suffix without further checks here?

Reply:
This seems reasonable. I will change the code so that if <local-connect> is present in the XML, then the connection string will be changed to prefix + globaltag + suffix, even if the prefix and suffix are empty or not specified in the XML. I will push a commit with this change soon.